plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java

```diff
@@ -184,6 +194,209 @@ public void testCommentViewColumn()
+        String nonExistedMetadataFileName = "00003-409702ba-4735-4645-8f14-09537cc0b2c8.metadata.json";
+        String tableLocation = "s3invalid://bucket-name/user/glue/warehouse/orders_5-581fad8517934af6be1857a903559d44";
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s', '%s')", schemaName, tableName, tableLocation, nonExistedMetadataFileName),
+                ".*Invalid location:.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", schemaName, tableName, tableLocation),
+                ".*Failed checking table's location:.*");
+    }
+
+    @Test
+    public void testRegisterTableWithInvalidParameters()
+    {
+        String tableName = "test_register_table_with_invalid_parameter_" + randomTableSuffix();
+        String tableLocation = schemaPath() + "/test/iceberg/glue/";
+
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s')", schemaName, tableName),
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', null)", schemaName),
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table (null, null)",
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s')", schemaName),
+                ".*'TABLE_NAME' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table (null)",
+                ".*'TABLE_NAME' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table ()",
+                ".*'SCHEMA_NAME' is missing.*");
+
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', null)", schemaName, tableName),
+                ".*table_location cannot be null.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', null, null)", schemaName),
+                ".*table_name cannot be null.*");
+        assertQueryFails("CALL iceberg.system.register_table (null, null, null)",
+                ".*schema_name cannot be null.*");
+
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", "", tableName, tableLocation),
+                ".*schemaName is empty.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", schemaName, "", tableLocation),
+                ".*tableName is empty.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", schemaName, tableName, ""),
+                ".*No versioned metadata file exists at location:.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s', '%s')", schemaName, tableName, tableLocation, ""),
+                ".*metadata_file_name cannot be empty when provided as an argument.*");
+    }
+
+    @Test
+    public void testRegisterTableWithInvalidMetadataFileName()
```

Comment:
Can you add a test which provides a `metadata_file_name` argument which is an existing file, but not the right kinda? One of the `avro` files, for example

Reply:
Do you mean to say that table location will have two different file formats?

Reply:
Yes, the Iceberg metadata has three different kinds of files in it, snapshots, manifest lists, and manifests. Manifest lists and manifests are avro files.

If a user provides one of those avro files as the file name, the procedure should fail.

Reply:
Got it.

Reply:
Added `testRegisterTableWithInvalidMetadataFile`